### ISSUE TITLE
Path resolving breaks on included @ sign

### ISSUE THREAD

**Issue by BeckmaR** [doxygen/doxygen#8186]
**Describe the bug**
When building on Jenkins and two builds are running at the same time, the second build's workspace path gets a `@2` appended. This seems to break doxygen:
```
/var/jenkins/workspace/PR-290@2/workspace/README.md:91: error: unable to resolve reference to '/var/jenkins/workspace/PR-290' for \ref command (warning treated as error, aborting now)
```
This results from a relative link in the readme which works fine without the `@`.

**Expected behavior**
doxygen resolves the file correctly

**To Reproduce**
Not possible since this happens on Jenkins

**Version**
1.8.19 selfbuilt

**Comment by albert-github** [comment]
When you write
> To Reproduce
> Not possible since this happens on Jenkins

this means that for us it will even be harder.

I do remember that recently something has been changed regarding the files names, so please try with the current master as well.

Furthermore providing the files:
- doxygen configuration settings
- README.md (at least the line where the reference is.
- the target of the reference

please provide:
- a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem and attach it to the issue? Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
A small note on the version 1.8.19, there is a newer version 1.8.20 available that was shortly released after 1.8.19 as there were some problem regarding the 1.8.19 version (see e.g. the bug fix section of the changelog for 1.8.20: (https://www.doxygen.nl/manual/changelog.html#log_1_8_20)).

**Comment by albert-github** [comment]
I was able to reproduce the problem (also with the current master: 1.9.0 (fa65bb38f81457d00f9c900bb57eb68bea59b1b4)).

With the input aa.md
```
The referenced file
```
and README.md:
```
readme please

[file ref](aa.md)
```
with the doxygen settings:
```
INPUT = ...\path@2
QUIET=YES
```

We see that the line:
```
[file ref](aa.md)
```
is translated into (using `doxygen -d markdown`):
```
@ref .../path@2/aa.md "file ref"
```
resulting in the warning:
```
.../path@2/README.md:3: warning: unable to resolve reference to '.../path' for \ref command
```
this is a similar warning as given in the issue.

It would be appreciated to see whether this is the similar setup as used for the Jenkins jobs so we are talking about the same problem.

**Comment by BeckmaR** [comment]
@albert-github I'm sorry I did not provide more specifics and I understand this creates more work for maintainers, sadly I did not have the time to create a complete example.
Also I mis-stated the version used, it was 1.8.20 and indeed I did test it with the master as well and it did not work for me either just like you found out. The setup you're describing matches the one on our jenkins very well and I'm sure it is the same issue, including the markdown link et cetera. 
Thank you for looking into this!

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8187

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
